inspect_image_test.go

```diff
@@ -248,49 +294,258 @@ func testInspectImage(t *testing.T, when spec.G, it spec.S) {
-					h.AssertNil(t, fakeImage.SetLabel(
-						"io.buildpacks.build.metadata",
-						`{
-  "processes": [
-    {
-      "type": "other-process",
-      "command": "/other/process",
-      "args": ["opt", "1"],
-      "direct": true
-    }
-  ]
-}`,
-					))
-
-					info, err := subject.InspectImage("some/image", useDaemon)
-					h.AssertNil(t, err)
+					when("ENTRYPOINT is set, but doesn't match an existing process", func() {
+						it.Before(func() {
+							mockDockerClient.EXPECT().ImageInspectWithRaw(gomock.Any(), fakeImage.Name()).
+								Return(types.ImageInspect{Config: &container.Config{Entrypoint: []string{EntrypointPrefix + "unknown-process"}}}, nil, nil)
+						})
+
+						it("returns nil default default process", func() {
+							info, err := subject.InspectImage("some/image", useDaemon)
+							h.AssertNil(t, err)
+
+							h.AssertEq(t, info.Processes,
+								ProcessDetails{
+									DefaultProcess: nil,
+									OtherProcesses: []launch.Process{
+										{
+											Type:    "other-process",
+											Command: "/other/process",
+											Args:    []string{"opt", "1"},
+											Direct:  true,
+										},
+										{
+											Type:    "web",
+											Command: "/start/web-process",
+											Args:    []string{"-p", "1234"},
+											Direct:  false,
+										},
+									},
+								},
+							)
+						})
+					})
 
-					h.AssertEq(t, info.Processes,
-						ProcessDetails{
-							DefaultProcess: nil,
-							OtherProcesses: []launch.Process{
-								{
-									Type:    "other-process",
-									Command: "/other/process",
-									Args:    []string{"opt", "1"},
-									Direct:  true,
+					when("ENTRYPOINT set to /cnb/lifecycle/launcher", func() {
+						it("returns a nil default process", func() {
+							mockDockerClient.EXPECT().ImageInspectWithRaw(gomock.Any(), fakeImage.Name()).
+								Return(types.ImageInspect{Config: &container.Config{Entrypoint: []string{LauncherEntrypoint}}}, nil, nil)
+
+							h.AssertNil(t, fakeImage.SetLabel(
+								"io.buildpacks.build.metadata",
+								`{
+					 "processes": [
+					   {
+					     "type": "other-process",
+					     "command": "/other/process",
+					     "args": ["opt", "1"],
+					     "direct": true
+					   }
+					 ]
+					}`,
+							))
+
+							info, err := subject.InspectImage("some/image", useDaemon)
+							h.AssertNil(t, err)
+
+							h.AssertEq(t, info.Processes,
+								ProcessDetails{
+									DefaultProcess: nil,
+									OtherProcesses: []launch.Process{
+										{
+											Type:    "other-process",
+											Command: "/other/process",
+											Args:    []string{"opt", "1"},
+											Direct:  true,
+										},
+									},
 								},
-							},
-						},
-					)
+							)
+						})
+					})
+
+					when("Inspecting Windows images", func() {
+						when(`ENTRYPOINT set to \cnb\lifecycle\launcher.exe`, func() {
```

Comment:
"\cnb\lifecycle\launcher.exe" seems to be misaligned with the test. The test uses `WindowsLauncherEntrypoint` which has the `c:` drive. Which one is the image actually expected to have?

We should probably not share production values for testing to prevent accidentally changing the tests.